FDB test fails on BSL setups on key error

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1. configure L2 on dut (https://github.com/Azure/SONiC/wiki/L2-Switch-mode)
2. run the fdb/test_fdb.py script on the dut

**Describe the results you received:**
```
duthosts = [<MultiAsicSonicHost arc-switch1004>]
rand_one_dut_hostname = 'arc-switch1004'
tbinfo = {'comment': 'Mellanox MTR testbed', 'conf-name': 'arc-switch1004-t0', 'duts': ['arc-switch1004'], 'duts_map': {'arc-switch1004': 0}, ...}

    @pytest.fixture(scope="module")
    def utils_vlan_intfs_dict_orig(duthosts, rand_one_dut_hostname, tbinfo):
        '''A module level fixture to record duthost's original vlan info
    
        Args:
            duthosts: All DUTs belong to the testbed.
            rand_one_dut_hostname: hostname of a random chosen dut to run test.
            tbinfo: A fixture to gather information about the testbed.
    
        Returns:
            VLAN info dict with original VLAN info
            Example:
                "Vlan1000": {
                    "fc02:1000::1/64": {},
                    "grat_arp": "enabled",
                    "proxy_arp": "enabled",
                    "192.168.0.1/21": {}
                }
        '''
        duthost = duthosts[rand_one_dut_hostname]
        cfg_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
        vlan_intfs_dict = {}
        for k, v in list(cfg_facts['VLAN'].items()):
            vlanid = v['vlanid']
>           for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
E           KeyError: 'VLAN_INTERFACE'

cfg_facts  = {'AUTO_TECHSUPPORT': {'GLOBAL': {'available_mem_threshold': '10.0', 'max_core_limit': '5.0', 'max_techsupport_limit': ...'Ethernet100': {'brkout_mode': '1x100G[50G,40G,25G,10G]'}, 'Ethernet104': {'brkout_mode': '2x50G[25G,10G]'}, ...}, ...}
duthost    = <MultiAsicSonicHost arc-switch1004>
duthosts   = [<MultiAsicSonicHost arc-switch1004>]
k          = 'Vlan1000'
rand_one_dut_hostname = 'arc-switch1004'
tbinfo     = {'comment': 'Mellanox MTR testbed', 'conf-name': 'arc-switch1004-t0', 'duts': ['arc-switch1004'], 'duts_map': {'arc-switch1004': 0}, ...}
v          = {'vlanid': '1000'}
vlan_intfs_dict = {}
vlanid     = '1000'

common/fixtures/duthost_utils.py:359: KeyError
```
This issue is similar to https://github.com/sonic-net/sonic-mgmt/issues/4742.

**Describe the results you expected:**
The test should not have any tests issues when running on BSL setups.

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

--- a/tests/common/fixtures/duthost_utils.py
+++ b/tests/common/fixtures/duthost_utils.py
@@ -342,6 +342,8 @@ def utils_vlan_intfs_dict_orig(duthosts, rand_one_dut_hostname, tbinfo):
     duthost = duthosts[rand_one_dut_hostname]
     cfg_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
     vlan_intfs_dict = {}
+    if 'VLAN_INTERFACE' not in cfg_facts:
+        return vlan_intfs_dict
     for k, v in cfg_facts['VLAN'].items():
         vlanid = v['vlanid']
         for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:

--- a/tests/common/helpers/portchannel_to_vlan.py
+++ b/tests/common/helpers/portchannel_to_vlan.py
@@ -281,6 +281,8 @@ def running_vlan_ports_list(duthosts, rand_one_dut_hostname, rand_selected_dut, 
     # key is dev name, value is list for configured VLAN member.
     for k, v in list(cfg_facts['VLAN'].items()):
         vlanid = v['vlanid']
+        if 'VLAN_INTERFACE' not in cfg_facts:
+            break
         for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
             # address could be IPV6 and IPV4, only need IPV4 here
             if addr and valid_ipv4(addr.split('/')[0]):
@@ -386,6 +388,11 @@ def setup_po2vlan(duthosts, ptfhost, rand_one_dut_hostname, rand_selected_dut, p
         return
 
     duthost = duthosts[rand_one_dut_hostname]
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    # Skip l2 mode
+    if "bgp_asn" not in cfg_facts["DEVICE_METADATA"]["localhost"]:
+        yield
+        return
     # --------------------- Setup -----------------------
     try:
         create_checkpoint(duthost, SETUP_ENV_CP)

--- a/tests/fdb/test_fdb.py
+++ b/tests/fdb/test_fdb.py
@@ -48,7 +48,13 @@ logger = logging.getLogger(__name__)
 
 
 @pytest.fixture(scope="module")
-def get_dummay_mac_count(tbinfo):
+def get_dummay_mac_count(tbinfo, duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    if "bgp_asn" not in cfg_facts["DEVICE_METADATA"]["localhost"]:
+        logger.info("Use dummy mac count {} for BSL test\n".format(DUMMY_MAC_COUNT_SLIM))
+        return DUMMY_MAC_COUNT_SLIM
+
     # t0-116 will take 90m with DUMMY_MAC_COUNT, so use DUMMY_MAC_COUNT_SLIM for t0-116 to reduce running time
     REQUIRED_TOPO = ["t0-116"]
     if tbinfo["topo"]["name"] in REQUIRED_TOPO:
@@ -205,8 +211,7 @@ def send_recv_eth(duthost, ptfadapter, source_ports, source_mac,                
         result = duthost.command("show mac", module_ignore_errors=True)
         logger.info("Dest MAC is {}, show mac results {}".format(dest_mac, result['stdout']))
         pytest_assert(False, "Expected packet was not received on ports {}"
-                             "Dest MAC in fdb is {}"
-                             .format(dest_ports, dest_mac.lower() in result['stdout'].lower()))
+                             .format(dest_ports))
 
 
 def setup_fdb(ptfadapter, vlan_table, router_mac, pkt_type, dummy_mac_count):

